src/Main.elm

```diff
@@ -103,10 +119,72 @@ update msg state =
                 , Cmd.none
                 )
 
+        AlphaSort header ->
+            let
+                sortType =
+                    setSortType header
+
+                direction =
+                    setDirection sortType state
+            in
+                ( { state
+                    | sortType = SortType direction sortType
+                  }
+                , Cmd.none
+                )
+
         _ ->
             ( state, Cmd.none )
 
 
+toggleFilter : String -> State a -> ( Bool, Header )
+toggleFilter header state =
+    let
+        headerAction =
+            setSortType header
+    in
+        case state.open of
+            ( True, hdr ) ->
+                if hdr == headerAction then
+                    ( False, headerAction )
+                else
+                    ( True, headerAction )
+
+            ( False, _ ) ->
+                ( True, headerAction )
```

Comment:
I think this could be condensed to

```elm
case state.open of
  (open, hdr ) ->
    (if hdr == headerAction then not open else open, headerAction)
```